src/NodaTime.Demo/OffsetDateTimeDemo.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright 2019 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NUnit.Framework;
+
+namespace NodaTime.Demo
+{    
+    public class OffsetDateTimeDemo
+    {
+        [Test]
+        public void Construction()
+        {
+            LocalDateTime localDateTime = new LocalDateTime(1985, 10, 26, 1, 18);
+            Offset offset = Offset.FromHours(-5);
+            OffsetDateTime offsetDateTime = Snippet.For(new OffsetDateTime(localDateTime, offset));
+
+            Assert.AreEqual(localDateTime, offsetDateTime.LocalDateTime);
+            Assert.AreEqual(offset, offsetDateTime.Offset);
+        }
+
+        [Test]
+        public void WithCalendar()
+        {
+            LocalDateTime localDateTime = new LocalDateTime(1985, 10, 26, 1, 18, CalendarSystem.Gregorian);
+            Offset offset = Offset.FromHours(-5);
+            OffsetDateTime original = new OffsetDateTime(localDateTime, offset);
+            OffsetDateTime updated = Snippet.For(original.WithCalendar(CalendarSystem.Iso));
+
+            Assert.AreNotEqual(original.LocalDateTime, updated.LocalDateTime);
```

Comment:
This won't be shown clearly in the snippet. (We have a special case for AreEqual, but not AreNotEqual). I suspect it's worth constructing the expected date instead. Then use

```csharp
Assert.AreEqual(new LocalDate(...).ToString('r'), updated.LocalDateTime.ToString('r'));
```

That will show the full round trip format, including the calendar. You could then remove the last line that compares the calendar explicitly.